internal/status/status.go

```diff
@@ -102,14 +103,14 @@ func (s *Status) Err() error {
 
 // WithDetails returns a new status with the provided details messages appended to the status.
 // If any errors are encountered, it returns nil and the first error encountered.
-func (s *Status) WithDetails(details ...proto.Message) (*Status, error) {
+func (s *Status) WithDetails(details ...protoV1.Message) (*Status, error) {
 	if s.Code() == codes.OK {
 		return nil, errors.New("no error details for status with code OK")
 	}
 	// s.Code() != OK implies that s.Proto() != nil.
 	p := s.Proto()
 	for _, detail := range details {
-		any, err := ptypes.MarshalAny(detail)
+		any, err := anypb.New(protoV1.MessageV2(detail))
```

Comment:
@menghanl 
> Seems what I was thinking was backwards? What's going to break is not `Details()`, but `WithDetails()`?
> 
> We cannot change the imported package from `"github.com/golang/protobuf/proto"` to `"google.golang.org/protobuf/proto"`,
> 
> because this changes the type `proto.Message` from `v1proto.Message` to `v2proto.Message`
> 
> thus it will change the function signature of `WithDetails()` (from `func(...v1proto.Message)` to `func(...v2proto.Message)`).
> 
> If a user is using very old generated code, the message types (e.g. `hellopb.Request`) will only implement `v1proto.Message` but not `v2proto.Message`. When they call `WithDetails(&hellopb.Request{})`, it will fail, because the types don't match.

I have updated this pull request. To keep backwards compatibility, the function signature of `WithDetails` is kept to `details ...protoV1.Message`. 

Changes from `ptypes` to `anypb` should still be transparent to end-users.

Reply:
With the discussions, I think this PR doesn't have enough benefits.
- There's no user noticeable change
  - especially that the users can use the status package with the new proto, so the status package doesn't stop the users from upgrading.
- The change switches from `ptypes` to `anypb`
  - but AFAICT, there's no visible difference between them, and it requires the conversion between v1 and v2, which makes it not as clean

Sorry for the back and forth. Let me know what you think.

Reply:
The initial intention of this PR was to remove the usage of [github.com/golang/protobuf](github.com/golang/protobuf) from the `status` package. However, doing so means that it is likely to break the code of some users since the function signature of `WithDetails` has been changed.

If it is infeasible for this change to be made at this time, then I am willing to close this PR.

Reply:
Right. Sorry again about that. I will close the PR. Progress on migrating the protobuf package is tracked in #3554